packages/scripts/src/tasks/providers/ios.ts

```diff
@@ -0,0 +1,59 @@
+import { envReplace } from "@idemsInternational/env-replace";
+import { Logger, generateVersionCode } from "../../utils";
+import { PATHS } from "shared";
+
+interface IiOSBuildOptions {
+  appId: string;
+  appName: string;
+  versionName: string;
+}
+
+/** Populate iOS template files with variables from deployment */
+const configure = async ({ appId, appName, versionName }: IiOSBuildOptions) => {
+  // TODO - allow user to input and update config where variables not defined (?)
+  if (!appId)
+    Logger.error({
+      msg1: `No appId configured for deployment`,
+      msg2: `Please set [ios.appId] in deployment config.ts`,
+    });
+  if (!appName)
+    Logger.error({
+      msg1: `No appName configured for deployment`,
+      msg2: `Please set [ios.appName] in deployment config.ts`,
+    });
+  if (!versionName)
+    Logger.error({
+      msg1: `No content version set configured for deployment`,
+      msg2: `Please set [git.content_tag_latest] in deployment config.ts`,
+    });
+  const versionCode = generateVersionCode(versionName);
+
+  // Populate templated iOS files
+  await envReplace.replaceFiles({
+    cwd: PATHS.ROOT_DIR,
+    // include both iOS folder and root (capacitor.config.ts)
+    includeFolders: ["ios/**", "."],
+    envAdditional: {
+      APP_ID: convertToValidIOSAppId(appId),
+      APP_NAME: appName,
+      VERSION_CODE: versionCode,
+      VERSION_NAME: versionName,
+    },
+    // As the default project.pbxproj file contains various variables env variables in ${...} format that
+    // are populated by xcode, target the specific variables that we've added into the template file
+    includeVariables: ["APP_ID", "APP_NAME", "VERSION_CODE", "VERSION_NAME"],
+  });
+};
+
+/**
+ * iOS app ID (aka "bundle ID") only supports alphanumeric characters (A–Z, a–z, and 0–9), hyphens (-), and periods (.),
+ * see https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleidentifier#discussion
+ * Replace underscores with hyphens to enable interoperability with Android app ID
+ * */
+function convertToValidIOSAppId(appId: string) {
```

Comment:
Good to know, although I think given that authors will be independently setting `android.app_id` and `ios.app_id` it's not unreasonable to expect different values to be put and just make it clear in the documentation, but equally no harm having the utility script here to allow platform conversion (should we also put the same in the android script to replace `-` with `_` ?)

I'm not sure how heavy-handed ios is when it comes to identifiers, last time I checked on Android play store once set an app id can't even be removed when replacing with another app, so may also want to emphasise that in future docs.
https://support.google.com/googleplay/android-developer/thread/203811370/how-i-can-delete-my-app-from-paly-console-account?hl=en

Reply:
I think my preference would be throwing an error when the iOS app ID does not follow the naming convention, rather than just allowing an incorrect name and dealing with it in the script. 

I can just imagine future-me or me-to-be's being confused why the iOS ID in the config does not correspond to the iOS ID in the Apple Store.

Reply:
Thanks Esmee, I think that's probably sensible on reflection. I've made a follow-up issue: #2268